File: My Bank App/ViewController.m
//
//  ViewController.m
//  My Bank App
//
//  Created by Arthur Walasek on 9/19/14.
//  Copyright (c) 2014 Callpod, Inc. All rights reserved.
//

#import "ViewController.h"
#import <KeeperExtensionSDK/KeeperExtensionSDK.h>
#import "AccountInformation.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    // A couple of different ways to show the keeper lock and activate the extension
//    [self.keeperLockBtn1 setHidden:![[KeeperSDK sharedExtension] isAppExtensionAvailable]];
//    [self.keeperLockBtn2 setHidden:![[KeeperSDK sharedExtension] isAppExtensionAvailable]];
    [self.keeperLockBtn1 setHidden:TRUE];
    [self.keeperLockBtn2 setHidden:TRUE];
    
    [self.keeperTabBtn setHidden:![[KeeperSDK sharedExtension] isAppExtensionAvailable]];
    
    self.txtUser.delegate = self;
    self.txtPassword.delegate = self;
}

- (void)viewWillAppear:(BOOL)animated
{
    NSString *user = @"";
    if ([AccountInformation sharedInstance].username) {
        user = [AccountInformation sharedInstance].username;
    }
    self.txtUser.text = user;
    self.txtPassword.text = @"";
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    [AccountInformation sharedInstance].username = self.txtUser.text;
    [AccountInformation sharedInstance].password = self.txtPassword.text;
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField
{
    [textField resignFirstResponder];
    return YES;
}

- (void)textFieldDidEndEditing:(UITextField *)textField {
}

- (IBAction)lockAction:(id)sender {
    __weak typeof (self) miniMe = self;
    [[KeeperSDK sharedExtension] findLoginForURLString:@"http://www.my-bank-website.com" forViewController:self sender:sender completion:^(NSDictionary *loginDict, NSError *error) {
        if (!loginDict) {
            if (error.code != AppExtensionErrorCodeCancelledByUser) {
                NSLog(@"Error invoking Keeper App Extension for find login: %@", error);
            }
            return;
        }
        
        __strong typeof(self) strongMe = miniMe;
        strongMe.txtUser.text = loginDict[AppExtensionUsernameKey];
        strongMe.txtPassword.text = loginDict[AppExtensionPasswordKey];
    }];
}

@end

File: My Bank App Swift/ViewController.swift
//
//  ViewController.swift
//  My Bank App Swift
//
//  Created by Jesse Gatt on 10/13/14.
//  Copyright (c) Keeper Security Inc. All rights reserved.
//

import UIKit
import KeeperExtensionSDK

class ViewController: UIViewController, UITextFieldDelegate {
    
    // WARNING: Clean up this code after testing
    @IBOutlet weak var txtUser: UITextField!
    @IBOutlet weak var txtPassword: UITextField!
    @IBOutlet weak var keeperLockBtn1: UIButton!
    @IBOutlet weak var keeperLockBtn2: UIButton!
    @IBOutlet weak var keeperTabBtn: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

        
        #if USE_KEEPER_TEXTFIELD_ICON
            keeperLockBtn1.hidden = !(KeeperSDK.sharedExtension().isAppExtensionAvailable())
            keeperLockBtn2.hidden = !(KeeperSDK.sharedExtension().isAppExtensionAvailable())
        #else
            keeperLockBtn1.hidden = true
            keeperLockBtn2.hidden = true
        #endif
        
        #if USE_KEEPER_TAB_ICON
            keeperTabBtn.hidden = !(KeeperSDK.sharedExtension().isAppExtensionAvailable())
        #else
            keeperTabBtn.hidden = true
        #endif
            
        
        
        txtUser.delegate = self
        txtPassword.delegate = self
        
    }
    
    
    override func viewDidAppear(animated: Bool) {
        
        if (AccountInformation.sharedInstance.username != nil) {
            txtUser.text = AccountInformation.sharedInstance.username
        }
        
        txtPassword.text = ""
        
        super.viewDidAppear(animated)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        AccountInformation.sharedInstance.username = txtUser.text
        AccountInformation.sharedInstance.password = txtPassword.text
    }
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        return true
    }
    
    func textFieldDidEndEditing(textField: UITextField) {
        
    }

    @IBAction func lockAction(sender: AnyObject) {
        
        KeeperSDK.sharedExtension().findLoginForURLString("http://www.my-bank-website.com", forViewController: self, sender: sender) { (loginDict:[NSObject : AnyObject]!, error:NSError!) -> Void in
            
            if (error != nil) {
                let newError = error as NSError!
                if (newError.code != AppExtensionErrorCodeCancelledByUser) {
                    println("Error invoking Keeper App Extension for find login: \(error)")
                }
                return;
                
            }
            
            if (loginDict != nil) {
                
                self.txtUser.text = loginDict[AppExtensionUsernameKey] as NSString
                self.txtPassword.text = loginDict[AppExtensionPasswordKey] as NSString
                
            }
            
        }
    }

}

